src/main/java/step04/domain/Ladder.java

```diff
@@ -0,0 +1,71 @@
+package step04.domain;
+
+import exception.InvalidCountOfStepsException;
+import strategy.PointStrategy;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class Ladder {
+    private final List<Step> ladder;
+    private final int countOfParticipants;
+
+    private Ladder(List<Step> ladder, int countOfParticipants) {
+        this.ladder = ladder;
+        this.countOfParticipants = countOfParticipants;
+    }
+
+    public static Ladder of(List<Step> ladder, int countOfParticipants) {
+        return new Ladder(ladder, countOfParticipants);
+    }
+
+    public static Ladder of(int countOfSteps, int countOfParticipants, PointStrategy pointStrategy) {
+        validate(countOfSteps);
+        List<Step> ladder = init(countOfSteps, countOfParticipants, pointStrategy);
+        return of(ladder, countOfParticipants);
+    }
+
+    private static List<Step> init(int countOfSteps, int countOfParticipants, PointStrategy pointStrategy) {
+        return Stream.generate(() -> Step.of(countOfParticipants, pointStrategy))
+                .limit(countOfSteps)
+                .collect(Collectors.toList());
+    }
+
+    private static void validate(int countOfSteps) {
+        if (countOfSteps < 1)
+            throw new InvalidCountOfStepsException();
+    }
+
+    public List<Step> getLadder() {
+        return ladder.stream()
+                .collect(Collectors.collectingAndThen(Collectors.toList(), Collections::unmodifiableList));
+    }
+
+    public TableOfResult matchTable() {
+        TableOfResult tableOfResult = Stream.iterate(0, n -> n + 1)
+                .limit(countOfParticipants)
+                .collect(Collectors.collectingAndThen(Collectors.toList(), TableOfResult::of));
+
+        for (int i = 0; i < ladder.size(); i++) {
+            tableOfResult = ladder.get(i).move(tableOfResult);
+        }
```

Comment:
사소하지만 향상된 for문을 사용한다면 더욱 가독성이 좋아질 것 같아요 😄 
```suggestion
        for (Step step : this.ladder) {
            tableOfResult = step.move(tableOfResult);
        }
```

Reply:
감사합니다! 향상된 for 문은 항상 습관적으로 제가 잘 못 써먹는 것 같아요 ㅠㅠ!!